gpsrenda/utils.py

```diff
@@ -0,0 +1,32 @@
+def extract_start_time(video_path):
+    import subprocess
+    from datetime import datetime
+    import pytz
+    from tzlocal import get_localzone
+
+    cmd = ["ffprobe",
```

Comment:
So one thing that this originally did is to extract the *precise* timestamp from a GoPro timecode.  It looks like `ffprobe` can do this too:

```
joshua@anima:~/gopro$ ffprobe -v quiet -print_format compact -select_streams d:0 -show_packets -show_data -show_entries packet=data GX010008.MP4 
packet|data=\n00000000: 0010 e104                                ....\n
```

The precise timecode is specified in frames since midnight.  Ideally, this would take the datecode from creation_time, and the timecode from the GoPro TCD metadata (if it's available).  Looks like the frame rate can be extracted reliably with:

```
joshua@anima:~/gopro$ ffprobe -v quiet -print_format compact -select_streams v:0 -show_streams -show_entries  stream=avg_frame_rate GX010008.MP4 
stream|avg_frame_rate=30000/1001|disposition:default=1|disposition:dub=0|disposition:original=0|disposition:comment=0|disposition:lyrics=0|disposition:karaoke=0|disposition:forced=0|disposition:hearing_impaired=0|disposition:visual_impaired=0|disposition:clean_effects=0|disposition:attached_pic=0|disposition:timed_thumbnails=0|tag:rotate=180|tag:creation_time=2021-05-30T10:15:09.000000Z|tag:language=eng|tag:handler_name= [...]
```

And finally, you can tell if something is a gopro file that has a gopro timecode with:

```
joshua@anima:~/gopro$ ffprobe -v quiet -print_format compact -select_streams d:0 -show_streams -show_entries stream=handler_name GX010008.MP4 
stream|disposition:default=1|disposition:dub=0|disposition:original=0|disposition:comment=0|disposition:lyrics=0|disposition:karaoke=0|disposition:forced=0|disposition:hearing_impaired=0|disposition:visual_impaired=0|disposition:clean_effects=0|disposition:attached_pic=0|disposition:timed_thumbnails=0|tag:creation_time=2021-05-30T10:15:09.000000Z|tag:language=eng|tag:handler_name=
                           GoPro TCD  |tag:timecode=10:35:44:04
```

Reply:
How much better is the exact timecode compared to the created time? Either way the GoPro and Garmin times tend to be off by a few seconds, so there is no escaping manually-aligning.